### PR TITLE
De Bruijn - dicho hybrid log2

### PR DESCRIPTION
**Pull request by QGarchery**
For science mainly ^^
hybrid-log2
Just one more gas than the best implementation ! But it probably would not be worth it anyways because of the added complexity of the code

### CODE REVIEW

**Comment by Rubilmax** [comment]
You know the bot is doing its job correctly!!! https://github.com/morpho-dao/morpho-utils/pull/94#issuecomment-1373473939

File: src/math/Math.sol [hunk 1]
```diff
@@ -26,45 +26,40 @@ library Math {
...
-
-            // This De Bruijn sequence begins with the byte 0, which is important to make shifting work like a rotation on the first byte
-            let deBruijnSeq := 0x00818283848586878898a8b8c8d8e8f929395969799a9b9d9e9faaeb6bedeeff
-            let key := shr(248, mul(x, deBruijnSeq)) // With the multiplication it works also for 0
-            y := shr(248, mload(add(m, key))) // Look in the table and take the first byte
+            let deBruijnSeq := 0x07c4acdd
+            let offset := shr(251, shl(224, mul(deBruijnSeq, x)))
+            y := add(
+                y,
+                shr(248, shl(mul(8, offset), 0x0009010a0d15021d0b0e10121619031e080c141c0f111807131b17061a05041f))
```
Comment (AtomicAzzaz) [inline src/math/Math.sol:61]:
```suggestion
                shr(248, shl(shl(3, offset), 0x0009010a0d15021d0b0e10121619031e080c141c0f111807131b17061a05041f))
```

Comment (AtomicAzzaz) [reply src/math/Math.sol:61]: 462

Comment (QGarchery) [reply src/math/Math.sol:61]: Reopening, maybe there are other optimizations on this

**Comment by MathisGD** [comment]
Let's close this one. It is linked to the #12 issue for future reference.

**Pull request closed without merging**